File: src/app/models/Box.ts
import { Product } from './Product';

export interface Box extends Product {
  motherboardFormFactor: string;  // ATX, mATX, ITX
  boxFormFactor: string;          // Mid Tower, Full Tower, Mini Tower
  color: string;
  maxGPULengthMM: number;
  maxCPUCoolerHeightMM: number;
  psuType: string;                // ATX, SFX
}

File: backend/src/main/java/bg/pcbudget/backend/models/Box.java
package bg.pcbudget.backend.models;

import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "boxes")
@Getter
@Setter
@NoArgsConstructor
public class Box extends Product {

  private String motherboardFormFactor;
  private String boxFormFactor;
  private String color;
  private Integer maxGPULengthMM;
  private Integer maxCPUCoolerHeightMM;
  private String psuType;
}
